packages/twitch-chat-client/src/ChatClient.ts

```diff
@@ -2112,6 +2094,19 @@ export class ChatClient extends IrcClient {
 		}
 	}
 
+	removeListener(): void;
+	removeListener(id: Listener): void;
+	removeListener(event: Function, listener?: Function): void;
+
+	removeListener() {
+		// Doing this with rest params would require a any[] type annotation which is bad and also
+		// would add a new function signature which is also not wanted thus using `arguments` makes more sense here.
```

Comment:
No, the implementation doesn't contribute to the actual signatures. But you don't have to use `any[]` either.

Please update the event emitter package (as a hard dependency in package.json! it's a necessary bugfix) and use its [new signatures](https://github.com/d-fischer/typed-event-emitter/blob/c513c13e6f969336147dec3aca61d1adbd047e1a/src/EventEmitter.ts#L27-L31) as a base for this. Also drop the `.call` from the super call - it's completely unnecessary.

To be honest, a complete fix for this would probably have to completely separate internal and external events. There are many `Promise`s that could leak because of calling this function while being connected. With separated events, event handlers wouldn't have to be recreated.

Reply:
I've updated the package and fixed the above code by using typescript type guards similar to the new implementation of `removeHandlers`.
I assume you meant `removeListener(idOrEvent?: Listener | Function, listener?: Function)` by "new signature" and that I should use that one but that signature isn't included with the newly generated type definitons: https://unpkg.com/@d-fischer/typed-event-emitter@3.0.2/lib/EventEmitter.d.ts so I can't use it to just forward the two parameters.

I'm sorry to say this but I currently don't have enough time to dig into the lib and understand how exactly events are passed so I'm not able to refactor this to use internal and external events.